src/gennylib/include/gennylib/PhaseLoop.hpp

```diff
@@ -100,6 +101,18 @@ class IterationChecker final {
             _rateLimiter =
                 phaseContext.workload().getRateLimiter(rateLimiterName, rateSpec.value());
         }
+
+        auto sleepBefore = phaseContext.get<TimeSpec, false>("SleepBefore").value_or(TimeSpec());
+        auto sleepAfter = phaseContext.get<TimeSpec, false>("SleepAfter").value_or(TimeSpec());
+
+        if ((sleepBefore || sleepAfter) && rateSpec) {
+            throw InvalidConfigurationException(
+                "Rate must *not* be specified alongside wither sleepBefore or sleepAfter"
+                "genny cannot enforce the global rate when there are mandatory sleeps in"
+                "each thread");
+        }
+        if (sleepBefore || sleepAfter)
+            _sleeper = std::make_unique<v1::Sleeper>(sleepBefore, sleepAfter);
```

Comment:
why have it be a ptr instead of a value? We don't need runtime indirection (virtuals). Our styleguide isn't very clear on this but imho it's cleaner and better to use values rather than owning ptrs unless you need the non-copyable/runtime-dispatch that the pointer provides.